model/model_ister.go

```diff
@@ -0,0 +1,327 @@
+		}
+		part.MountPoint = curr.Mount
+
+		partitions[curr.Partition] = part
+		disks[curr.Disk] = partitions
+	}
+
+	// For each disk, add the partitions as children elements
+	for _, curr := range si.TargetMedias {
+		partitions := disks[curr.Name]
+		var children = make([]*storage.BlockDevice, len(partitions), len(partitions))
+		i := 0
+		for _, part := range partitions {
+			e := storage.BlockDevice{}
+			e = part
+			children[i] = &e
+			i++
+		}
+		curr.Children = children           // Set Children
+		curr.Name = "${" + curr.Name + "}" // Update Name
+		si.AddTargetMedia(curr)            // Set TargetMedia
+	}
+
+	// Set Version. Any non numeric version (e.g. "latest") is set to 0
+	v, err := ic.Version.Int64()
+	if err == nil {
+		si.Version = uint(v)
+	} else {
+		si.Version = 0
+	}
+
+	// Remove the first occurrence of a kernel bundle and set it as the Kernel
+	for _, curr := range ic.Bundles {
+		if strings.HasPrefix(curr, "kernel-") {
+			if si.Kernel == nil {
+				si.Kernel = &kernel.Kernel{Bundle: curr}
+				continue
+			}
+		}
+		si.Bundles = append(si.Bundles, curr) // Set Bundles
+	}
+
+	// Set Users
+	for _, curr := range ic.Users {
+		u := user.User{}
+		u.Login = string(curr.UID)
+		u.UserName = curr.Username
+		u.Admin = curr.Sudo
+		u.SSHKeys = append(u.SSHKeys, curr.Key)
+		si.Users = append(si.Users, &u)
+	}
+
+	// Set KernelArguments
+	if ic.Cmdline != "" {
+		s := strings.Split(ic.Cmdline, " ")
+		ka := kernel.Arguments{}
+		ka.Add = s
+		si.KernelArguments = &ka
+	}
+
+	// Set Hostname
+	si.Hostname = ic.Hostname
+
+	// Set NetworkInterfaces
+	if ic.StaticIP != nil {
+		nw := network.Interface{}
+		nw.Name = ic.StaticIP.Iface
+		nw.Gateway = ic.StaticIP.Gateway
+		nw.DHCP = false //  always static IP
+
+		s := strings.Split(ic.StaticIP.Address, "/")
+		addr := network.Addr{}
+		addr.IP, addr.NetMask = s[0], s[1]
+		nw.Addrs = append(nw.Addrs, &addr)
+
+		si.NetworkInterfaces = append(si.NetworkInterfaces, &nw)
+	}
+
+	// Set PostInstall with PostNonChroot
+	for _, curr := range ic.PostNonChroot {
+		pi := InstallHook{Chroot: false, Cmd: curr}
+		si.PostInstall = append(si.PostInstall, &pi)
+	}
+	// Set PostInstall with PostNonChrootShell  prefixed with "/bin/bash -c "
+	for _, curr := range ic.PostNonChrootShell {
+		pi := InstallHook{Chroot: false, Cmd: "/bin/bash -c " + curr}
+		si.PostInstall = append(si.PostInstall, &pi)
+	}
+
+	si.LegacyBios = ic.LegacyBios // Set LegacyBios
+	si.SwupdMirror = ic.MirrorURL // Set SwupdMirror
+
+	si.HTTPSProxy = ic.HTTPSProxy // Set HTTPSProxy
+	if si.HTTPSProxy == "" {
+		si.HTTPSProxy = ic.HTTPProxy
+		fmt.Println("WARNING: Mapping HTTPProxy in json to HTTPSProxy in yaml")
+	} else {
+		fmt.Println("WARNING: Skipping HTTPProxy mapping")
+	}
+
+	// Hardcoding the missing required fields
+	si.Telemetry = &telemetry.Telemetry{Enabled: false}              // Set Telemetry
+	si.Keyboard = &keyboard.Keymap{Code: "us"}                       // Set Keyboard
```

Comment:
Can we switch the default keyboard to keyboard.DefaultKeyboard

Reply:
Done